dali/test/python/test_RN50_data_fw_iterators.py

```diff
@@ -0,0 +1,177 @@
+                                            std=[0.229 * 255,0.224 * 255,0.225 * 255])
+        self.coin = ops.CoinFlip(probability=0.5)
+
+    def define_graph(self):
+        rng = self.coin()
+        jpegs, labels = self.input(name="Reader")
+        images = self.decode_gpu(jpegs)
+        images = self.res(images)
+        output = self.cmnp(images.gpu(), mirror=rng)
+        return (output, labels.gpu())
+
+parser = argparse.ArgumentParser(description='Test RN50 augmentation pipeline with different FW iterators')
+parser.add_argument('-g', '--gpus', default=1, type=int, metavar='N',
+                    help='number of GPUs (default: 1)')
+parser.add_argument('-b', '--batch', default=2048, type=int, metavar='N',
+                    help='batch size (default: 2048)')
+parser.add_argument('-p', '--print-freq', default=10, type=int,
+                    metavar='N', help='print frequency (default: 10)')
+parser.add_argument('-j', '--workers', default=3, type=int, metavar='N',
+                    help='number of data loading workers (default: 3)')
+parser.add_argument('--prefetch', default=2, type=int, metavar='N',
+                    help='prefetch queue deptch (default: 2)')
+parser.add_argument('--fp16', action='store_true',
+                    help='Run fp16 pipeline')
+parser.add_argument('--nhwc', action='store_true',
+                    help='Use NHWC data instead of default NCHW')
+parser.add_argument('-i', '--iters', default=-1, type=int, metavar='N',
+                    help='Number of iterations to run (default: -1 - whole data set)')
+parser.add_argument('-e', '--epochs', default=1, type=int, metavar='N',
+                    help='Number of epochs to run (default: 1)')
+args = parser.parse_args()
+
+N = args.gpus             # number of GPUs
+BATCH_SIZE = args.batch   # batch size
+LOG_INTERVAL = args.print_freq
+WORKERS = args.workers
+PREFETCH = args.prefetch
+FP16 = args.fp16
+NHWC = args.nhwc
+
+print("GPUs: {}, batch: {}, workers: {}, prefetch depth: {}, loging interval: {}, fp16: {}, NHWC: {}"
+      .format(N, BATCH_SIZE, WORKERS, PREFETCH, LOG_INTERVAL, FP16, NHWC))
+
+class AverageMeter(object):
+    """Computes and stores the average and current value"""
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self.val = 0
+        self.avg = 0
+        self.sum = 0
+        self.count = 0
+        self.avg_last_n = 0
+        self.max_val = 0
+
+    def update(self, val, n=1):
+        self.val = val
+        self.max_val = max(self.max_val, val)
+        self.sum += val * n
+        self.count += n
+        self.avg = self.sum / self.count
+
+import nvidia.dali.plugin as nvidia_plugin
+from nvidia.dali.plugin.mxnet import DALIClassificationIterator as MXNetIterator
+from nvidia.dali.plugin.pytorch import DALIClassificationIterator as PyTorchIterator
+from nvidia.dali.plugin.tf import DALIIterator as TensorFlowIterator
```

Comment:
Maybe `imports` to the top of the file? They are not conditionally imported, so they could be grouped together.

Reply:
I am not sure I understand. Can you elaborate?

Reply:
What I meant is that canonical way for Python scripts would be to have all `imports` at the top of the file (like `PEP8` advises). Sometimes something might be conditionally imported (like inside an `if` statement) and then it makes sense to have `imports` somewhere in the middle of the file.

Reply:
will do